packages/dd-trace/test/setup/mocha.js

```diff
@@ -74,10 +76,20 @@ function withNamingSchema (
           })
         })
 
+        beforeEach(async () => {
+          await new Promise(resolve => setTimeout(resolve, 30))
```

Comment:
What's this about?

Reply:
I was encountering a data race where the mock agent was still handling traces from previous test cases while this test case was being started.

Reply:
We'll need to isolate the cause of these race conditions instead of adding the slowdown. Maybe some day the CI servers will get slower and we'll get confusing errors that we don't know how to debug.

Is it specific tests that fail? Or is it random? It could be that a subset of tests are marked as complete even though they're still doing some background tasks.